### PR TITLE
Fixed all remaining clippy lints (except the complexity ones)

### PR DESCRIPTION
**Pull request by therealprof**
Signed-off-by: Daniel Egger <daniel@eggers-club.de>

### CODE REVIEW

**Comment by jamesmunns** [comment]
Hm. Tests passed but Bors died :(

bors retry

**Comment by jamesmunns** [comment]
Hey @japaric, it looks like bors is having some issues here. Builds are passing, but bors is cancelling the merge.

**Comment by jamesmunns** [comment]
Update: Somehow I have bors control here, but I am not a codeowner, so I can't approve the review. Sorry for all the noise. We'll have to wait for @rust-embedded/tools.

File: src/generate/device.rs [hunk 1]
```diff
@@ -100,21 +105,19 @@ pub fn render(d: &Device, target: &Target, nightly: bool, device_x: &mut String)
...
     out.extend(interrupt::render(target, &d.peripherals, device_x)?);
 
-    let core_peripherals: &[&str];
-
-    if fpu_present {
-        core_peripherals = &[
-            "CBP", "CPUID", "DCB", "DWT", "FPB", "FPU", "ITM", "MPU", "NVIC", "SCB", "SYST", "TPIU"
-        ];
+    let core_peripherals = if fpu_present {
+        vec![
```
Comment (japaric) [inline src/generate/device.rs]: why did this become a vec? You can write, for example:

``` rust
let x: &[_] = if true {
    &[0, 1]
} else {
    &[2, 3, 4]
};
```

File: src/generate/device.rs [hunk 2]
```diff
@@ -49,7 +54,7 @@ pub fn render(d: &Device, target: &Target, nightly: bool, device_x: &mut String)
         });
     }
 
-    match *target {
+    match target {
```
Comment (japaric) [inline src/generate/device.rs]: Just a comment: this opts into the "match ergonomics" feature which may raise the MSRV -- not that it was ever documented or guaranteed.

File: src/generate/device.rs (hunk 1 above)
Comment (therealprof) [reply src/generate/device.rs]: Was recommended by clippy. Dunno. I can undo it.

Comment (therealprof) [reply src/generate/device.rs]: Oh, I remember. rustc complained about not being able to assign arrays with different lengths.

Comment (therealprof) [reply src/generate/device.rs]: TIL I can assign arrays of various lengths by declaring the type as `&[_]`. How odd. :-D

**Pull request merged**